base/db/R/get.new.site.R

```diff
@@ -0,0 +1,145 @@
+        if (is.null(site$id) | is.na(site$id)) {
+            PEcAn.logger::logger.warn("Site dataframe does not have an id column. Generating a unique ID")
+            site.id <- generate_new_siteID()
+            PEcAn.logger::logger.info(paste0("Generated siteID:", site.id))
+            if ((!is.null(site$lat) && !is.null(site$lon)) |
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                new.site <- data.frame(
+                    id = as.numeric(site.id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(site$lat, "-", site$lon)
+            } else {
+                new.site <- data.frame(
+                    id = site.id,
+                    lat = NULL,
+                    lon = NULL
+                )
+                str_ns <- paste0(new.site$id %/% 1e+09, "-", new.site$id %% 1e+09)
+                # We used a different str_ns as identifier here due to absence of lat/lon
+            }
+        } else {
+            # setup site database number, lat, lon and name and copy for format.vars if new input
+            if ((!is.null(site$lat) && !is.null(site$lon)) |
+                (!is.na(site$lat) && !is.na(site$lon))
+            ) {
+                new.site <- data.frame(
+                    id = as.numeric(site$id),
+                    lat = site$lat,
+                    lon = site$lon
+                )
+                str_ns <- paste0(site$lat, "-", site$lon)
+            } else {
+                latlon <- query.site(site$id, con = con)[c("lat", "lon")]
+                new.site <- data.frame(
+                    id = as.numeric(site$id),
+                    lat = latlon$lat,
+                    lon = latlon$lon
+                )
+                str_ns <- paste0(new.site$lat, "-", new.site$lon)
+            }
+        }
+    }
+
+    site.info <- list(new.site = new.site, str_ns = str_ns)
+
+    return(site.info)
+}
+
+# Function to generate a normalized site ID using lat / lon
+# Example usage
+# siteID <- generate_siteID(-34.9284989, 138.6007456)
+generate_siteID <- function(lat, lon) {
+    # Normalize latitude and longitude
+    norm_lat <- lat + 90 # Shift range to [0, 180]
+    norm_lon <- lon + 180 # Shift range to [0, 360]
+
+    # Scale normalized values (Assuming 4 digits each for lat and lon)
+    scaled_lat <- as.integer((norm_lat / 180) * 9999) # Scaled latitude
+    scaled_lon <- as.integer((norm_lon / 360) * 9999) # Scaled longitude
+
+    # Create a unique ID by concatenating scaled values
+    siteID <- as.integer(paste0(scaled_lat, scaled_lon))
+
+    # Return the siteID
+    return(siteID)
+}
+
+# Function to generate a new siteID (db-less runs ONLY)
+generate_new_siteID <- function() {
+    # Generate a random number. Assuming higher order integers to increase randomness in IDs
+    random_id <- sample(10000:99999999, 1)
+    return(as.numeric(random_id))
+}
```

Comment:
One approach I suggest is as follow : Hashing to generate a unique id using lat and lon of the site. This would ensure greater precision upto 8 decimal points.